Fill.py

```diff
@@ -64,26 +65,45 @@ def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations,
                     placed_item = location.item
                     # Unplaceable items can sometimes be swapped infinitely. Limit the
                     # number of times we will swap an individual item to prevent this
-                    if swapped_items[placed_item.player, placed_item.name] > 0:
+                    swap_count = swapped_items[placed_item.player,
+                                               placed_item.name]
+                    if swap_count > 1:
```

Comment:
This lets us try each location an additional time. VVVVVV 4 cost doors occasionally needed more tries to find a valid solution.